bin/inx-chronicle/src/api/v2/routes.rs

```diff
@@ -0,0 +1,641 @@
+    let end_milestone = end_milestone(&database, end_timestamp).await?;
+
+    let mut query_doc =
+        vec![doc! { "$match": { "milestone_index": { "$gt": start_milestone, "$lt": end_milestone } } }];
+
+    if let Some(address) = address.as_ref() {
+        query_doc.extend([
+            doc! { "$match": { "message.payload.data.essence.data.outputs.address.data": address } },
+            doc! { "$set": {
+                "message.payload.data.essence.data.outputs": {
+                    "$filter": {
+                        "input": "$message.payload.data.essence.data.outputs",
+                        "as": "output",
+                        "cond": { "$eq": [ "$$output.address.data", address ] }
+                    }
+                }
+            } },
+        ]);
+    }
+    query_doc.extend([
+        doc! { "$unwind": { "path": "$message.payload.data.essence.data.outputs", "includeArrayIndex": "message.payload.data.essence.data.outputs.idx" } },
+        doc! { "$sort": { "milestone_index": -1 } },
+        doc! { "$skip": (page_size * page) as i64 },
+        doc! { "$limit": page_size as i64 },
+    ]);
+    if *included {
+        query_doc[0]
+            .get_document_mut("$match")?
+            .insert("inclusion_state", LedgerInclusionState::Included as u8 as i32);
+    }
+
+    let outputs = database
+        .collection::<Document>("stardust_messages")
+        .aggregate(query_doc, None)
+        .await?
+        .try_collect::<Vec<_>>()
+        .await?;
+
+    Ok(OutputsForQueryResponse {
+        query,
+        max_results: page_size,
+        count: outputs.len(),
+        output_ids: outputs
+            .into_iter()
+            .map(|mut rec| {
+                let mut payload = rec.take_path("message.payload.data")?.to_document()?;
+                let transaction_id = TransactionId::from_str(payload.get_str("transaction_id")?)?;
+                let idx = payload
+                    .take_path("essence.data.outputs")?
+                    .to_document()?
+                    .get_as_u16("idx")?;
+                let output_id = OutputId::new(transaction_id, idx)?;
+                Ok(if expanded {
+                    let inclusion_state = rec
+                        .get_as_u8("inclusion_state")
+                        .ok()
+                        .map(LedgerInclusionState::try_from)
+                        .transpose()?;
+                    let milestone_index = rec.get_i32("milestone_index").ok().map(|m| m as u32);
+                    Record {
+                        id: output_id.to_string(),
+                        inclusion_state,
+                        milestone_index,
+                    }
+                    .into()
+                } else {
+                    output_id.to_string().into()
+                })
+            })
+            .collect::<Result<_, APIError>>()?,
+    })
+}
+
+async fn transaction_history(
```

Comment:
Is that part of `/api/v2` or should we think about a separate Chronicle API?

Reply:
It could be separate.